sdk/clientcore/core/src/main/java/io/clientcore/core/utils/CoreUtils.java

```diff
@@ -447,6 +446,39 @@ public static String appendQueryParam(String url, String key, String value) {
         }
     }
 
+    /**
+     * Appends the query parameter list values to the given host URL.
+     *
+     * @param url the base URL to which the query parameter will be appended.
+     * @param key the name of the query parameter (e.g., "api-version", "name", "After").
+     * @param value the value of the query parameter.
+     * @param delimiter The delimiter to use if the value is a list.
+     * @return the updated URL with the appended query parameter.
+     */
+    public static String appendMultiQueryParam(String url, String key, List<?> value, char delimiter) {
+        if (url == null || key == null || value == null) {
+            return url;
+        }
+
+        StringBuilder newUrl = new StringBuilder(url);
+        if (!url.contains("?")) {
+            newUrl.append("?");
+        } else if (!url.endsWith("&") && !url.endsWith("?")) {
+            newUrl.append("&");
+        }
```

Comment:
Creating a new StringBuilder instance and checking for `?` can be expensive in a hot path. We should collect all the query params that should be appended and append them in one in one go. `appendQueryParams(String baseUrl, Map<String, QueryParam> queryParams)`. This method should return the final URL with all query params appended.

Reply:
Discussed offline, updated the method usage to Map<String, Object>.